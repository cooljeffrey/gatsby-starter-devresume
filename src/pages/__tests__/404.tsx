import React from "react";
import renderer from "react-test-renderer";
import NotFoundPage from "../404";
import * as Gatsby from "gatsby";

describe("NotFoundPage", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
  useStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        author: "Simon",
        description: "Resume description",
        title: "Resume Title",
      },
    },
  }));

  it("renders correctly", () => {
    const tree = renderer.create(<NotFoundPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
