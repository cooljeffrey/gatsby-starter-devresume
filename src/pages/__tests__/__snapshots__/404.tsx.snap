// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<div
  className="main-wrapper"
>
  <div
    className="container px-3 px-lg-5"
  >
    <article
      className="resume-wrapper mx-auto theme-bg-light p-5 mb-5 my-5 shadow-lg"
    >
      <h1>
        404: Not Found
      </h1>
      <p>
        You just hit a route that doesn't exist... the sadness.
      </p>
    </article>
  </div>
  <footer
    className="footer text-center py-4"
  >
    <small
      className="copyright text-muted"
    >
      Designed with 
      <i
        className="fas fa-heart"
      />
       by
       
      <a
        className="theme-link"
        href="http://themes.3rdwavemedia.com"
        rel="noreferrer"
        target="_blank"
      >
        Xiaoying Riley
      </a>
       
      for developers
    </small>
  </footer>
</div>
`;
